core/state_transition.go

```diff
@@ -424,6 +437,16 @@ func (st *stateTransition) execute() (*ExecutionResult, error) {
 	if st.gasRemaining < gas {
 		return nil, fmt.Errorf("%w: have %d, want %d", ErrIntrinsicGas, st.gasRemaining, gas)
 	}
+	// Gas limit suffices for the floor data cost (EIP-7623)
+	if rules.IsPrague {
+		floorDataGas, err = FloorDataGas(msg.Data)
+		if err != nil {
+			return nil, err
+		}
+		if st.gasRemaining < floorDataGas {
```

Comment:
nitpick, it's probably better to use `msg.GasLimit` rather than the gasRemaining